dns_server/src/config.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use clap::Parser;
+use directories::UserDirs;
+
+// TODO: Allow using with testnet
+
+#[derive(Parser, Debug)]
+pub struct DnsServerConfig {
+    /// The path to the data directory
+    #[clap(long, default_value_t = default_data_dir())]
+    pub datadir: String,
+
+    /// IP address and UDP port to listen on
+    #[clap(long, default_values_t = vec!["[::]:53".to_string()])]
+    pub bind_addr: Vec<String>,
+
+    /// Optional list of initial node addresses to connect
+    #[clap(long)]
+    pub add_node: Vec<String>,
+
+    /// Hostname of the DNS seed
+    #[clap(long)]
+    pub host: String,
+
+    /// Hostname of the nameserver
+    #[clap(long)]
+    pub ns: String,
+
+    /// E-Mail address reported in SOA records (RNAME)
```

Comment:
Well, I'd be too hesitant to enforce using a valid email address... I'm always worried about spam.

Reply:
Looks like the SOA record must always be present: https://serverfault.com/a/1024024
And `rname` is the admin's email address in the SOA.
It's possible to run DNS without the SOA record, but all of Bitcoin's public seed DNSs have it, so I decided to make it mandatory.

Reply:
I would want to keep it optional and use a bogus email address if none is provided. This isn't the 90s anymore.

Can we technically make it optional?

Reply:
Sure, I made both the `nameserver` and `mbox` arguments optional (the same way it's done in bitcoin-seeder).